src/bottom-sheet/bottom-sheet.ts

```diff
@@ -0,0 +1,210 @@
+import { html, LitElement, PropertyValues, nothing } from 'lit';
+import { property, query, state } from 'lit/decorators.js';
+import '@tapsioss/icons/dist/icons/cross';
+
+export class BottomSheet extends LitElement {
+  static readonly shadowRootOptions = {
+    ...LitElement.shadowRootOptions,
+    delegatesFocus: true,
+  };
+
+  @property({ type: Boolean, reflect: true, attribute: 'is-open' })
+  isOpen: boolean = false;
+
+  @property({ type: Boolean, reflect: true })
+  isDismissible: boolean = true;
+
+  @property({ type: Boolean, reflect: true, attribute: 'has-dimmer' })
+  hasDimmer: boolean = false;
+
+  @property({ type: String, reflect: true })
+  title: string = '';
+
+  @property({ type: Boolean, reflect: true, attribute: 'is-expanded' })
+  isExpanded: boolean = false;
+
+  @property({ type: Boolean, reflect: true })
+  showGrabber: boolean = true;
+
+  @state() private touchDirection: string = '';
+
+  @state() private disappear = false;
+
+  @state() private hasSlotHeaderContent = false;
+
+  private startX: number = 0;
+  private startY: number = 0;
+
+  @query('#bottom-sheet')
+  private bottomSheetElement?: HTMLElement | null;
+
+  @query('.bottom-sheet-header')
+  private headerElement?: HTMLElement | null;
+
+  @query('.bottom-sheet-body')
+  private bodyElement?: HTMLElement | null;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.showGrabber && this.addEventListeners();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.showGrabber && this.removeEventListeners();
+  }
+
+  private addEventListeners = (): void => {
+    this.addEventListener('touchstart', this.handleTouchStart);
+    this.addEventListener('touchend', this.handleTouchEnd);
+  };
+
+  private removeEventListeners = (): void => {
+    this.removeEventListener('touchstart', this.handleTouchStart);
+    this.removeEventListener('touchend', this.handleTouchEnd);
+  };
+
+  protected updated(changed: PropertyValues): void {
+    if (changed.has('showGrabber')) {
+      this.showGrabber &&
+        this.style.setProperty('--tap-bottom-sheet-header-padding', '12px');
+    }
```

Comment:
Don't we have to add/remove event listeners in case the `showGrabber` changes?

Reply:
'showGrabber' controls the visibility of the grabber element at the top of the bottom sheet header, maintaining a **consistent value** for each bottom sheet in app.